core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -48,12 +48,14 @@ class LogManager(val logDirs: Array[File],
                  val cleanerConfig: CleanerConfig,
                  ioThreads: Int,
                  val flushCheckMs: Long,
-                 val flushCheckpointMs: Long,
+                 val flushRecoveryOffsetCheckpointMs: Long,
+                 val flushStartOffsetCheckpointMs: Long,
```

Comment:
No, we intentionally only checkpoint logStartOffset. Is there any reason to checkpoint lowWatermark?